docs(CONTRIBUTE): add note about warnings when running `yarn build`. Add sections

gr2m opened the pull request:
Follow up to https://github.com/pikapkg/snowpack/discussions/958#discussioncomment-58808

## Changes

- add note about warnings when running `yarn build`
- add sections

## Testing

n/a

gr2m committed:
docs(CONTRIBUTE): add note about warnings when running `yarn build`. Add sections

gr2m commented on CONTRIBUTING.md:
```diff
@@ -1,41 +1,57 @@
-### Discussion
+## Discussion
 
 [Join the Pika Discord](https://discord.gg/rS8SnRk)
```
It's out of scope of this pull request, but I wonder if we should add a link to the repository's discussion forum here?

FredKSchott replied:
yea! We still see more people hitting our issues page than our "discussions page" by almost 2x

melissamcewen commented on CONTRIBUTING.md:
```diff
@@ -1,41 +1,57 @@
 
+## Requirements
+
+You need [yarn](https://classic.yarnpkg.com/en/docs/install) to setup the `snowpack` repository locally on your machine
```
Might want to note you can use another similar package manager like npm if you want (I think??). I'd imagine most people who could contribute know that, but just in case someone thinks you absolutely need yarn.

FredKSchott replied:
We actually use yarn's workspaces feature to manage the multiple packages in the monorepo, so in this case you actually do need yarn. @gr2m it would be good to explain that quickly here, in case others have questions or assume it might just be optional.

FredKSchott replied:
Maybe:

"Snowpack uses [yarn workspaces](https://classic.yarnpkg.com/) to manage the multiple packages contianed in this repo. To contribute, [make sure that you have Yarn installed on your machine.]((https://classic.yarnpkg.com/en/docs/install)

FredKSchott commented on CONTRIBUTING.md:
```diff
@@ -1,41 +1,57 @@
-# Run your tests
 yarn build
 yarn --force
 yarn test
```
add note here about running single test: `yarn test -t treeshake` 
or multiple, related tests: `yarn test -t config`

gr2m replied:
done via 982ff14

gr2m committed:
Update CONTRIBUTING.md

Co-authored-by: Fred K. Schott <fkschott@gmail.com>

gr2m committed:
docs(CONTRIBUTING): filter tests that are being run using `--testNamePattern`

/cc @FredKSchott follow up to https://github.com/pikapkg/snowpack/pull/979\#discussion_r481633835

FredKSchott commented on CONTRIBUTING.md:
````diff
@@ -8,38 +8,60 @@ Please note that all activity on the [`pikapkg/snowpack` repository](https://git
+You can filter the tests that are being run using Jest's [`--testNamePattern`](https://jestjs.io/docs/en/cli#--testnamepatternregex) CLI option. You can ignore the `123 snapshots obsolete` messages.
+
+```bash
+yarn test --testNamePattern treeshake
````
nit: I'd love a mention somewhere here that `-t` is also supported. Something like:

> You can filter the tests that are being run using Jest's `--testNamePattern` (alias: `-t`) CLI option.

drwpow commented:
@gr2m feel free to merge this when you’re ready!

gr2m committed:
mention `-t` alias for `jest --testNamePattern`

resolves https://github.com/pikapkg/snowpack/pull/979\#discussion_r483395738
